File: PHP Excel Examples/WorksheetsApiDemo/sortRange.php
<?php

use Spire\Cloud\Excel\Sdk\Configuration;
use Spire\Cloud\Excel\Sdk\Model\DataSorter;
use Spire\Cloud\Excel\Sdk\Model\SortKey;

$appId = "your id";
$appKey = "your key";
$baseUrl = "https://api.e-iceblue.cn";
$configuration = new Configuration($appId, $appKey, $baseUrl);
$apiInstance = new WorksheetsApi($configuration);

$name = "SortRange.xlsx";
$folder = "input";
$storage = null;
$sheetName = "Sheet4";
$cellArea = "A1:G10";
$dataSorter = new DataSorter();
$dataSorter->setCaseSensitive(true);
$dataSorter->setHasHeaders(true);
$dataSorter->setSortLeftToRight(false);
$SortKey = new SortKey();
$SortKey->setKey(0);
#supports sort-orders:Ascending/Descending/Top
$SortKey->setSortOrder("Ascending");
#supports sort-types:Values/BackgroundColor/FontColor
$SortKey->setSortType( "Values");
$SortKey1 = new SortKey();
$SortKey1->setKey(1);
$SortKey1->setSortOrder("Descending");
$SortKey1->setSortType( "Values");
$KeyList = array($SortKey,$SortKey1);
$dataSorter->setKeyList($KeyList);
$result = $apiInstance->sortRange($name, $sheetName, $cellArea, $dataSorter, $folder, $storage);

File: Python Excel Examples/WorksheetsApiDemo/sortRange.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.worksheets_api.WorksheetsApi(configuration)

name = "SortRange.xlsx"
folder = "ExcelDocument";
storage = ""
sheetName = "Sheet4"
cellArea = "A1:G10"
dataSorter = spirecloudexcel.models.DataSorter()
dataSorter.case_sensitive = True
dataSorter.has_headers = True
dataSorter.sort_left_to_right = False
KeyList = list()
KeyList.append(spirecloudexcel.models.SortKey(0, "Ascending", "Values"))
KeyList.append(spirecloudexcel.models.SortKey(1, "Descending", "Values"))
KeyList.append(spirecloudexcel.models.SortKey(2, "Top", "BackgroundColor"))
KeyList.append(spirecloudexcel.models.SortKey(3, "Top", "FontColor"))
dataSorter.key_list = KeyList
api.sort_range(name, sheetName, cellArea, dataSorter, folder=folder, storage=storage)
